pkg/metrics/server.go

```diff
@@ -63,23 +66,27 @@ func stopServer(srv *http.Server) {
 	}
 }
 
+func (Server) Start(ctx context.Context) error {
+	return RunServer(MetricsPort, ctx)
+}
+
 // RunServer starts the server, and watches the tlsCert and tlsKey for certificate changes.
-func RunServer(port int, stopCh <-chan struct{}) {
+func RunServer(port int, ctx context.Context) error {
```

Comment:
Can you please specify where are we calling this method?

Reply:
called here:
https://github.com/openshift/cluster-node-tuning-operator/blob/d534c86bfb48e2245dd4556e2bb2c1b2aa1119ff/pkg/metrics/server.go#L69
As a runnable here: 
https://github.com/openshift/cluster-node-tuning-operator/blob/d534c86bfb48e2245dd4556e2bb2c1b2aa1119ff/cmd/cluster-node-tuning-operator/main.go#L123
Started here:
https://github.com/openshift/cluster-node-tuning-operator/blob/d534c86bfb48e2245dd4556e2bb2c1b2aa1119ff/cmd/cluster-node-tuning-operator/main.go#L140

Reply:
oh, it's a nice one.

Reply:
You can pass the metrics to the manager: https://github.com/operator-framework/operator-sdk/blob/master/testdata/go/v3/memcached-operator/main.go#L70

By using SDK/Kubebuilder scaffolds you won that by free as well. See: https://book.kubebuilder.io/reference/metrics.html